tensorflow-core/tensorflow-core-api/pom.xml

```diff
@@ -0,0 +1,321 @@
+            <compilerOption>${project.basedir}/src/main/native/eager_operation_jni.cc</compilerOption>
+            <compilerOption>${project.basedir}/src/main/native/eager_session_jni.cc</compilerOption>
+            <compilerOption>${project.basedir}/src/main/native/exception_jni.cc</compilerOption>
+            <compilerOption>${project.basedir}/src/main/native/graph_jni.cc</compilerOption>
+            <compilerOption>${project.basedir}/src/main/native/graph_operation_builder_jni.cc</compilerOption>
+            <compilerOption>${project.basedir}/src/main/native/graph_operation_jni.cc</compilerOption>
+            <compilerOption>${project.basedir}/src/main/native/saved_model_bundle_jni.cc</compilerOption>
+            <compilerOption>${project.basedir}/src/main/native/server_jni.cc</compilerOption>
+            <compilerOption>${project.basedir}/src/main/native/session_jni.cc</compilerOption>
+            <compilerOption>${project.basedir}/src/main/native/tensorflow_jni.cc</compilerOption>
+            <compilerOption>${project.basedir}/src/main/native/tensor_jni.cc</compilerOption>
+            <compilerOption>${project.basedir}/src/main/native/utils_jni.cc</compilerOption>
+          </compilerOptions>
+        </configuration>
+        <executions>
+          <execution>
+            <id>javacpp-validate</id>
+            <phase>validate</phase>
+            <goals>
+              <goal>build</goal>
+            </goals>
+          </execution>
+          <execution>
+            <id>javacpp-build</id>
+            <phase>initialize</phase>
+            <goals>
+              <goal>build</goal>
+            </goals>
+            <configuration>
+              <skip>${javacpp.build.skip}</skip>
+              <buildCommand>
+                <program>bash</program>
+                <argument>${project.basedir}/build.sh</argument>
+              </buildCommand>
+              <workingDirectory>${project.basedir}</workingDirectory>
+            </configuration>
+          </execution>
+          <execution>
+            <id>javacpp-clean</id>
+            <phase>clean</phase>
+            <goals>
+              <goal>build</goal>
+            </goals>
+            <configuration>
+              <skip>${javacpp.build.skip}</skip>
+              <buildCommand>
+                <program>bazel</program>
+                <argument>clean</argument>
+              </buildCommand>
+              <workingDirectory>${project.basedir}</workingDirectory>
+            </configuration>
+          </execution>
+          <execution>
+            <id>javacpp-parser</id>
+            <phase>generate-sources</phase>
+            <goals>
+              <goal>parse</goal>
+            </goals>
+            <configuration>
+              <skip>${javacpp.parser.skip}</skip>
+              <outputDirectory>${project.basedir}/src/gen/java</outputDirectory>
+              <classOrPackageName>org.tensorflow.c_api.presets.*</classOrPackageName>
```

Comment:
The TF Java code is based upon the Google Java Style Guide, which states that package names [shouldn't have underscores in it](https://google.github.io/styleguide/javaguide.html#s5.2.1-package-names)...

Can we rename that package to something else, like `nativeapi`?

(BTW would be great that generated class names also sticks to camelcase, with no underscores, but I guess that could be a challenge to change this, right? e.g `TFEContext` instead of `TFE_Context`, or `DeallocatorPointerLongPointer` instead of `Deallocator_Pointer_long_Pointer`)

Reply:
Well, we can start customizing names like that yes, but the idea with JavaCPP is to map the underlying API to something that's as close as possible to the native API. If we start renaming things randomly like that, it increases the chances of conflicts and it makes it harder to understand the mapping. Besides, the C API is very hard to use, it's not meant to be used by end users, and making it even harder to use isn't going to help. :)

Reply:
Since the classes won't be public, I'm ok to just rename the package as well.

Reply:
My thought was that having a package name like `c_api` makes it less likely for the casual to want to use it. I also thought about `capi`, but I just don't find it as obvious as `c_api`, which is used by TensorFlow in the docs making it clearer that it's related: https://www.tensorflow.org/install/lang_c

Reply:
Ok then

Reply:
Maybe let's get the opinion of at least one other person? In a lot of places where JavaCPP just uses `_`, I know Panama uses `$` because, you know, `$` isn't a valid character in C/C++ identifiers, but it is in Java!! (I wonder how they are going to get this passed the CSR group...) @Craigacp Any opinions about this?

Reply:
I think for internal package names which we won't expose to the user then `_` is fine. Panama uses `$` for internally generated things because it guarantees no conflicts, but in this case the header file is actually called `c_api.h`, so mirroring the name is fine. Panama's `jextract` gives it the same name (see here https://hg.openjdk.java.net/panama/dev/raw-file/c359a9e944de/doc/panama_foreign.html#using-tensorflow-c-api-mac-os).